sql/src/main/java/io/crate/execution/engine/window/WindowBatchIterator.java

```diff
@@ -182,16 +219,44 @@ private void computeAndFillStandaloneOutgoingCellsFor(Row sourceRow) {
         }
     }
 
-    private void collectPeers() {
-        Iterable<Row> rows = peerRows.stream().collect(collector);
-        peerRows.clear();
-        peersCollected = true;
-        Row collectedRow = rows.iterator().next();
-        collectorNumberOfColumns = collectedRow.numColumns();
-        for (int i = 0; i < collectorNumberOfColumns; i++) {
-            outgoingCells[i] = collectedRow.get(i);
+    private void executeWindowFunctions() {
+        WindowFrame currentFrame = new WindowFrame(
+            windowRowPosition,
+            sourceRowsConsumed,
+            InMemoryBatchIterator.of(windowForCurrentRow, SentinelRow.SENTINEL)
```

Comment:
We could probably pass in the list or an iterable instead of a `BatchIterator`. It looks like currently the users of the `WindowFrame` aren't properly consuming the `BatchIterator` (they don't call loadNextBatch - so they only work due to the implementation detail of using an InMemoryBatchIterator).